docs/browser/events.md

```diff
@@ -0,0 +1,158 @@
+|domainLookupStart | long | || Recommended |
+|domainLookupEnd | long | || Recommended |
+|connectStart | long | || Recommended |
+|secureConnectionStart | long | || Recommended |
+|connectEnd | long | || Recommended |
+|requestStart | long | || Recommended |
+|responseStart | long | || Recommended |
+|responseEnd | long | || Recommended |
+
+## Exception
+
+The event name MUST be `browser.exception`.
+
+This event describes an error or exception that occurs in a browser application.
+
+The following table describes the payload fields that MUST be used to describe the details of event.
+
+| Body Field  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `exception.file_name` | string | Name of the file that generated the error | `foo.js` | Recommended |
+| `exception.line_number` | int | Line number where the error occurred |  | Recommended |
+| `exception.column_number` | int | Column number where the error occurred |  | Recommended |
+| `exception.message` | string | The exception message. | `Division by zero`; `Can't convert 'int' object to str implicitly` | Recommended |
+| `exception.stacktrace` | string | A stacktrace as a string in the natural representation for the language runtime. The representation is to be determined and documented by each language SIG. | `Exception in thread "main" java.lang.RuntimeException: Test exception\n at com.example.GenerateTrace.methodB(GenerateTrace.java:13)\n at com.example.GenerateTrace.methodA(GenerateTrace.java:9)\n at com.example.GenerateTrace.main(GenerateTrace.java:5)` | Recommended |
+| `exception.type` | string | The type of the exception (its fully-qualified class name, if applicable). The dynamic type of the exception should be preferred over the static type in languages that support it. | `java.net.ConnectException`; `OSError` | Recommended |
+
+## UserAction
+
+The event name MUST be`user_action`.
+
+This event describes actions performed by the user such as click, scroll, zoom, resize, etc.
+
+The following table describes the payload fields that MUST be used to describe the details of event.
+
+| Body Field  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `element` | string | Target element tag name (obtained via `event.target.tagName`) | `button` | Recommended |
+| `element_xpath` | string | Target element xpath | `//*[@id="testBtn"]` | Recommended |
+| `user_action_type` | string | Type of interaction. See enum [here](https://github.com/microsoft/ApplicationInsights-JS/blob/941ec2e4dbd017b8450f2b17c60088ead1e6c428/extensions/applicationinsights-clickanalytics-js/src/Enums.ts) for potential values we could add support for. | `click` | Required |
+| `click_coordinates` | string | Click coordinates captured as a string in the format {x}X{y}.  Eg. 345X23 | `345X23` | Recommended |
+| `tags` | string[] | Grab data from data-otel-* attributes in tree | `[data-otel-asd="value" -> asd attr w/ "value"]` | Recommended |
```

Comment:
sounds like a template attribute type (e.g. check out http headers as an  example)

Reply:
@lmolkova I guess you are referring to `http.request.header.<key>` and that would apply if we were to name the field as `data-otel-<key>` - however, instead, we are naming the field as `tags` of type `string array`. hence not using the template attribute type.

Reply:
are there reasons not to use template type and not call tags something else? E.g.  

- `tag.<key>=value`
- `data-otel-<key>=value`

What's written here now results in key-value-pairs being serialized to string (and you don't  even describe the format. What I guess you want is an array of key value pairs.

You may try document `tags` type as array of objects and explain that each object is a key value pair where key is the name of `data-otel-*` and value is the corresponding value, but template attributes would give you the same outcome.

Reply:
@lmolkova I converted to tag.<key> format.